### PR TITLE
fix #78, handle permissions for docker.sock in a better way

### LINKED ISSUE #78: [gitkubed] Goes to CrashLoopBackOff on Docker for Desktop
`start_sshd.sh` script needs some modifications due to the difference in ownership of `docker.sock` on Docker for Mac.

State of Gitkube on Docker for Windows in unknown. Any help in identifying the status is appreciated.

### PR DESCRIPTION
**Pull request by shahidhk**
@tirumaraiselvan This PR fixes #78. But, while testing some other issue with certain parameters in `sshd_config` being deprecated has appeared. Please check once before merging.

The following lines appear continuously with different port numbers in the logs for gitkubed:
```
rexec line 19: Deprecated option KeyRegenerationInterval
rexec line 20: Deprecated option ServerKeyBits
rexec line 31: Deprecated option RSAAuthentication
rexec line 38: Deprecated option RhostsRSAAuthentication
Could not load host key: /etc/ssh/ssh_host_dsa_key
Did not receive identification string from 10.1.0.1 port 43146
```

### CODE REVIEW

File: build/gitkubed/start_sshd.sh [hunk 1]
```diff
@@ -1,12 +1,16 @@
 #!/usr/bin/env sh
```
Comment (tirumaraiselvan) [inline build/gitkubed/start_sshd.sh]: You can remove `/hasura-data/group` mount from the deployment.

**Comment by tirumaraiselvan** [comment]
To remove the deprecated messages in the logs, just remove the deprecated options from sshd_config

```
sed -i '/KeyRegenerationInterval/d' /etc/ssh/sshd_config
sed -i '/ServerKeyBits/d' /etc/ssh/sshd_config
sed -i '/RSAAuthentication/d' /etc/ssh/sshd_config
sed -i '/RhostsRSAAuthentication/d' /etc/ssh/sshd_config
```

**Comment by tirumaraiselvan** [comment]
Also comment the line `HostKey /etc/ssh/ssh_host_dsa_key` to remove the host_dsa_key line

**Comment by shahidhk** [comment]
@tirumaraiselvan To remove the deprecated messages, you've mentioned `/etc/ssh/sshd_config`. But aren't we using `/sshd-lib/sshd_config`?

**Comment by tirumaraiselvan** [comment]
@shahidhk Right. You just need to remove it from here https://github.com/hasura/gitkube/blob/master/build/gitkubed/sshd_config

**Comment by shahidhk** [comment]
I figured.

**Comment by shahidhk** [comment]
@tirumaraiselvan Any idea about `Did not receive identification string from 172.17.0.1 port 58866`?

**Pull request merged**